stdlib/public/runtime/ImageInspection.h

```diff
@@ -30,6 +30,18 @@ namespace swift {
     void *symbolAddress;
   };
 
+#if defined(__ELF__) || defined(__ANDROID__)
+
+struct SectionInfo {
+  uint64_t size;
+  const char *data;
+};
+
+// Called by injected constructors when a dynamic library is loaded.
+void addNewDSOImage(const void *addr);
```

Comment:
After #6820 it's spelled `SWIFT_RUNTIME_EXPORT`. That handles all of the mangling and visibility fuss required to provide a symbol that is called by outside code.